api/commands/reminders/remind.js

```diff
@@ -41,34 +39,52 @@ module.exports = class RemindCommand extends Command {
   }
 
   run(msg, { target, content, datetime }) {
-    const author = {
-      username: msg.message.author.username,
-      discriminator: msg.message.author.discriminator
-    };
+    const { author } = msg.message;
 
-    let parsedTime;
+    let timezone;
+    let reminders = [];
+    let remindersJson;
 
     if (!chrono.parseDate(datetime)) {
       return new Promise((resolve, reject) =>
         reject(msg.say(exceptions.invalid_datetime_format)));
     }
 
-    return selectTz([author.username, author.discriminator])
-    .then((timezone) => {
-      if (!timezone) {
-        return Promise.reject(msg.say(exceptions.timezone_not_set));
+    return tzStore.getAsync(author.id)
+    .then((tz) => {
+      if (!tz) {
+        msg.say(exceptions.timezone_not_set);
+      } else {
+        timezone = tz;
       }
-
-      parsedTime = parseDate(datetime, timezone);
+    })
+    .then(() => {
+      const parsedTime = parseDate(datetime, timezone);
 
       if (parsedTime.delayAmt < 500) {
         return Promise.reject(msg.say(exceptions.past_time));
```

Comment:
just a note about a future improvement/idea:
it seems to be misleading to say <500 ms is in the past when say i make a reminder 1 second from now and latency is trash (or even 500ms from now)
could consider telling the user to make a reminder at least 1 second from now or maybe hard enforce a minimum future time as a reminder 1 second from now is stupid
thoughts?